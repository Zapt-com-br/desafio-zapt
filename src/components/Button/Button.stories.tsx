import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import Button, { ButtonProps } from '.';

export default {
  title: 'Generic/Button',
  component: Button,
} as Meta;

export const ButtonDefault: Story<ButtonProps> = args => <Button {...args} />;

ButtonDefault.args = {
  label: 'Criar um grupo',
};
