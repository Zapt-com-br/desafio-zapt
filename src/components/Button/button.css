.btn-main {
  min-width: 161px;
  height: 51px;
  cursor: pointer;
  color: #fff;
  border: none;
  border-radius: 4px;
  background-color: var(--blue);
  font-size: 14px;
}