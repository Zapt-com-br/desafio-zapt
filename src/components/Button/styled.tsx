import styled from 'styled-components'

export const Button = styled.button`
  border: none;
  background-color: #334bc8;
  border-radius: 4px;
  color: #fff;

  padding: 1rem 2rem;

  transition: background-color 0.4s;

  &:hover {
    background: #ff5b58;
  }
`
