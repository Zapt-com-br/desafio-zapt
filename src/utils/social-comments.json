{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
