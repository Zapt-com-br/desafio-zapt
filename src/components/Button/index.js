import React from 'react';
import './button.css';

function Button({ title }) {
  return (
    <button className="btn-main">{title}</button>
  );
}

export default Button;