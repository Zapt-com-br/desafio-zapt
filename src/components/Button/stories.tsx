import { Story } from '@storybook/react'
import {
  BoxArrowLeft,
  CheckCircle,
  ExclamationTriangle,
  GearFill,
  InfoCircle,
  PencilSquare,
  Save,
  Trash,
  X,
  XCircle
} from '@styled-icons/bootstrap'
import styled from 'styled-components'
import Button, { BtnProps } from '.'

const WrapperBtn = styled.div`
  display: flex;
  gap: 2rem;
  flex-wrap: wrap;
  justify-content: center;
`

export const Template: Story<BtnProps> = args => (
  <WrapperBtn>
    <Button {...args} />
    <Button {...args} icon={<Save />} />
    <Button {...args} text="" icon={<GearFill />} />
    <Button {...args} text="" icon={<Save />} />
    <Button {...args} text="" icon={<PencilSquare />} />
    <Button {...args} text="" icon={<Trash />} />
    <Button {...args} text="" icon={<X />} />
    <Button {...args} text="" icon={<InfoCircle />} />
    <Button {...args} text="" icon={<ExclamationTriangle />} />
    <Button {...args} text="" icon={<CheckCircle />} />
    <Button {...args} text="" icon={<XCircle />} />
    <Button {...args} text="" icon={<BoxArrowLeft />} />
  </WrapperBtn>
)

export const Default = Template.bind({})
