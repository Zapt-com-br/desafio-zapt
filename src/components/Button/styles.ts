import styled, { css, DefaultTheme } from 'styled-components'
import { darken } from 'polished'
import { ButtonProps } from '.'

type WrapperProps = Pick<ButtonProps, 'color' | 'fullWidth'>

const wrapperModifiers = {
  primary: (theme: DefaultTheme) => css`
    background: ${theme.colors.primaryLight};

    &:hover {
      background: ${darken(0.05, theme.colors.primaryLight)};
    }
  `,

  secondary: (theme: DefaultTheme) => css`
    background: ${theme.colors.secondary};

    &:hover {
      background: ${darken(0.05, theme.colors.secondary)};
    }
  `,

  gray: (theme: DefaultTheme) => css`
    background: ${theme.colors.gray};

    &:hover {
      background: ${darken(0.05, theme.colors.gray)};
    }
  `,

  fullWidth: () => css`
    width: 100%;
  `
}

export const Wrapper = styled.button<WrapperProps>`
  ${({ theme, color, fullWidth }) => css`
    display: inline-flex;
    align-items: center;
    justify-content: center;
    padding: ${theme.spacings.base} ${theme.spacings.lg};
    height: 5.1rem;

    border: 0;
    border-radius: ${theme.border.radius};
    color: ${theme.colors.white};
    cursor: pointer;
    font-size: ${theme.font.sizes.small};
    text-decoration: none;

    ${!!color && wrapperModifiers[color](theme)}
    ${!!fullWidth && wrapperModifiers.fullWidth()}
  `}
`
