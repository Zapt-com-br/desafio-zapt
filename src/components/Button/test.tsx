import { render, screen } from 'utils/test-utils'

import Button from '.'

describe('<Button />', () => {
  it('should render the button with style and children', () => {
    render(<Button>Crie um grupo</Button>)

    expect(
      screen.getByRole('button', { name: /Crie um grupo/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /Crie um grupo/i })).toHaveStyle({
      'font-size': '1.4rem',
      'font-weight': '400',
      color: '#FFFFFF',
      'background-color': '#334BC8',
      padding: '1.6rem 3.2rem 1.8rem'
    })
  })
})
