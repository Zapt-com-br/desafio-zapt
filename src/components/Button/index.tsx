import React, { ButtonHTMLAttributes } from 'react';
import * as Styled from './styles';

import { Typography } from 'components';

export type ButtonProps = {
  label: string;
  onClick?: () => void;
} & ButtonHTMLAttributes<HTMLButtonElement>;

const Button = ({ label, onClick }: ButtonProps) => (
  <Styled.Wrapper onClick={onClick}>
    <Typography color="white" size="small">
      {label}
    </Typography>
  </Styled.Wrapper>
);

export default Button;
