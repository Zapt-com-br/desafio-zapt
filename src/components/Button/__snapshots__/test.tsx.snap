// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render primary button 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.6rem 2.4rem;
  height: 5.1rem;
  border: 0;
  border-radius: 0.4rem;
  color: #ffffff;
  cursor: pointer;
  font-size: 1.4rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #334bc8;
}

.c0:hover {
  background: #2e43b4;
}

<button
  class="c0"
  color="primary"
>
  Criar um grupo
</button>
`
